### PR TITLE
Rds integration

### LINKED ISSUE #559: Long term solution for dbs and third party dependencies
Per our discussion here: https://github.com/Originate/exosphere/issues/508, we are going with a more permanent solution to third party dependencies in production. What do we think about this proposed format in `application.yml` taken from the discussion in #508:
```
development:
  dependencies:
    - name: exocom
      version: 0.24.0
    - name: mongo
      version: 3.4.0
       silent: true
       config:
          ...

production:
  dependencies
    - name: exocom
      version: 0.24.0
    - name: mongo
      version: 3.4.0
      config:
         ...
```

### PR DESCRIPTION
**Pull request by hugoesthere**
The end of a long journey to resolve: https://github.com/Originate/exosphere/issues/559

### CODE REVIEW

File: src/config/rds_production_dependency.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+type rdsProductionDependency struct {
+	config    types.ProductionDependencyConfig
+	appConfig types.AppConfig
+}
+
+// GetDockerConfig returns docker configuration and an error if any
+func (r *rdsProductionDependency) GetDockerConfig() (types.DockerConfig, error) {
+	return types.DockerConfig{
+		Image: fmt.Sprintf("%s:%s", r.config.Name, r.config.Version),
```
Comment (hugoesthere) [inline src/config/rds_production_dependency.go]: Wondering what we should do here as we still have the un-needed step where a DB dependency's docker image is uploaded to ECR

File: src/types/production_dependency_config_options.go [hunk 2]
```diff
@@ -2,7 +2,12 @@ package types
 
 // ProductionDependencyConfigOptions represents the configuration of a development dependency
 type ProductionDependencyConfigOptions struct {
-	//TODO: populate with RDS fields
+	AllocatedStorage string `yaml:"allocated-storage,omitempty"`
+	InstanceClass    string `yaml:"instance-class,omitempty"`
+	DbName           string `yaml:"db-name,omitempty"`
+	Username         string `yaml:",omitempty"`
+	PasswordEnvVar   string `yaml:"password-env-var,omitempty"` //TODO: verify this is set via exo configure
+	StorageType      string `yaml:"storage-type,omitempty"`
```
Comment (charlierudolph) [inline src/types/production_dependency_config_options.go]: Weren't we going to group this in an RDS config?

File: src/config/rds_production_dependency.go (hunk 1 above)
Comment (charlierudolph) [reply src/config/rds_production_dependency.go]: Can we update the ProductionDependency interface to have something like `HasDockerConfig` or something? And the builder doesn't add those when they return false. 

Alternatively, we could make it so a return value from this function of nil means a similar thing. We may need to update the return type to be a pointer to allow nil to be returned

File: src/types/production_dependency_config_options.go (hunk 2 above)
Comment (hugoesthere) [reply src/types/production_dependency_config_options.go]: Right, I'll add that in

File: src/config/rds_production_dependency.go (hunk 1 above)
Comment (hugoesthere) [reply src/config/rds_production_dependency.go]: What about generic dependencies? They would all either have to be pushed to ECR or not. Should we add yet another field to `ProductionDependencyConfig`?

Comment (charlierudolph) [reply src/config/rds_production_dependency.go]: Aren't generic dependencies currently based on a docker image (which means we would always need to push)

Comment (hugoesthere) [reply src/config/rds_production_dependency.go]: I'm not even sure what a generic dependency is at this point. I guess we'll just always push for now and then add a field if it becomes necessary in the future.

**Comment by hugoesthere** [comment]
@charlierudolph should be good for another look

File: src/config/rds_production_dependency.go (hunk 1 above)
Comment (charlierudolph) [reply src/config/rds_production_dependency.go]: Can this now be updated to return an empty docker config?

**Review by charlierudolph — APPROVED** [review]

**Pull request merged**